### ISSUE TITLE
UselessParentheses false positive with bitwise not operator (`~`)

### ISSUE THREAD

**Issue by spaze** [slevomat/coding-standard#1752]
I have a method that contains only this body (for testing):

```php
	private function foo(): bool
	{
		~(1 - 1);
	}
```

PHPCS reports the following on `~(1 - 1);`:
```
----------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------
 11 | ERROR | [x] Useless parentheses. (SlevomatCodingStandard.PHP.UselessParentheses.UselessParentheses)
----------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------------------
```

Running phpcbf then removes the outer parentheses so that the method body reads

```php
~1 - 1;
```

But those two lines don't give the same result:

```php
> ~(1 - 1);
= -1

> ~1 - 1;
= -3
```

Using slevomat/coding-standard 8.18.0, phpcs 3.13.0.

**Comment by spaze** [comment]
Wow, that was quick! Thanks a lot @kukulich!
